src/common/polyfill_thread.h

```diff
@@ -0,0 +1,299 @@
+// SPDX-FileCopyrightText: 2022 yuzu Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <version>
+
+#ifdef __cpp_lib_jthread
+
+#include <stop_token>
+#include <thread>
+
+namespace Common {
+
+template <typename Condvar, typename Lock, typename Pred>
+void CondvarWait(Condvar& cv, Lock& lock, std::stop_token token, Pred&& pred) {
+    cv.wait(lock, token, std::move(pred));
+}
+
+} // namespace Common
+
+#else
+
+#include <atomic>
+#include <functional>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <thread>
+
+namespace std {
+
+typedef list<function<void()>> __internal_stop_state_callbacks;
```

Comment:
```suggestion
using __internal_stop_state_callbacks = list<function<void()>>;
```

Also since prefixed double underscores are reserved for implementation use, the underscores should be removed, since we don't control that.

Reply:
Since I am acting as the implementation in this case, am I not allowed to do this? Otherwise it will be visible unprefixed in std namespace.

Reply:
No, because there's already an existing overarching implementation this is working on, so this can break if any of its symbols change and happen to clash (which can happen in the future).

It's best to leave it unprefixed and inside a `detail` namespace to communicate it's an internal header detail (which is a fairly normal convention).